apps/price_pusher/src/evm/evm.ts

```diff
@@ -188,31 +206,40 @@ export class EvmPricePusher implements IPricePusher {
       throw e;
     }
 
-    let gasPrice = Number(
-      (await this.customGasStation?.getCustomGasPrice()) ||
-        (await this.web3.eth.getGasPrice())
-    );
+    const fees = await this.publicClient.estimateFeesPerGas();
+
+    this.logger.debug({ fees }, "Estimated fees");
+
+    let gasPrice =
+      Number(await this.customGasStation?.getCustomGasPrice()) ||
+      Number(fees.gasPrice) ||
+      Number(fees.maxFeePerGas);
 
     // Try to re-use the same nonce and increase the gas if the last tx is not landed yet.
     if (this.pusherAddress === undefined) {
-      this.pusherAddress = (await this.web3.eth.getAccounts())[0];
+      this.pusherAddress = this.walletClient.account!.address;
     }
+
     const lastExecutedNonce =
-      (await this.web3.eth.getTransactionCount(this.pusherAddress)) - 1;
+      (await this.publicClient.getTransactionCount({
+        address: this.pusherAddress,
+      })) - 1;
 
     let gasPriceToOverride = undefined;
 
     if (this.lastPushAttempt !== undefined) {
       if (this.lastPushAttempt.nonce <= lastExecutedNonce) {
         this.lastPushAttempt = undefined;
       } else {
-        gasPriceToOverride = Math.ceil(
-          this.lastPushAttempt.gasPrice * this.overrideGasPriceMultiplier
-        );
+        gasPriceToOverride =
```

Comment:
the ceil may have been important here -- does it cause a problem if the result here isn't an integer?

Reply:
oh is this caught already by the call to `.round` below? Maybe this is fine.